File: accudrop/app/src/main/java/me/chrislane/accudrop/viewmodel/RouteViewModel.java
package me.chrislane.accudrop.viewmodel;

import android.arch.lifecycle.LiveData;
import android.arch.lifecycle.MutableLiveData;
import android.arch.lifecycle.ViewModel;
import android.location.Location;

import java.util.List;

public class RouteViewModel extends ViewModel {

    private final MutableLiveData<List<Location>> route = new MutableLiveData<>();
    private final MutableLiveData<Integer> minAltitude = new MutableLiveData<>();
    private final MutableLiveData<Integer> maxAltitude =  new MutableLiveData<>();

    /**
     * Get the recommended route.
     *
     * @return A <code>LiveData</code> object containing the route.
     */
    public LiveData<List<Location>> getRoute() {
        return route;
    }

    /**
     * Set the route.
     *
     * @param route The route to be set.
     */
    public void setRoute(List<Location> route) {
        this.route.setValue(route);
    }

    /**
     * Set a point in the route.
     *
     * @param index The index position in the route to set.
     * @param point The point to set in the route.
     */
    public void setPoint(int index, Location point) {
        List<Location> route = this.route.getValue();
        if (route != null) {
            route.set(index, point);
        }
    }

    public LiveData<Integer> getMinAltitude() {
        return minAltitude;
    }

    public LiveData<Integer> getMaxAltitude() {
        return maxAltitude;
    }

    public void setMinAltitude(int minAltitude) {
        this.minAltitude.setValue(minAltitude);
    }

    public void setMaxAltitude(int maxAltitude) {
        this.maxAltitude.setValue(maxAltitude);
    }
}

File: app/src/main/java/me/chrislane/accudrop/viewmodel/RouteViewModel.kt
package me.chrislane.accudrop.viewmodel

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import android.location.Location

import com.google.android.gms.maps.model.LatLng

class RouteViewModel : ViewModel() {

    private val route = MutableLiveData<MutableList<Location>>()
    private val minAltitude = MutableLiveData<Int>()
    private val maxAltitude = MutableLiveData<Int>()
    private val target = MutableLiveData<LatLng>()

    /**
     * Get the recommended route.
     *
     * @return A `LiveData` object containing the route.
     */
    fun getRoute(): LiveData<MutableList<Location>> {
        return route
    }

    /**
     * Set the route.
     *
     * @param route The route to be set.
     */
    fun setRoute(route: MutableList<Location>) {
        this.route.value = route
    }

    fun getMinAltitude(): LiveData<Int> {
        return minAltitude
    }

    fun getMaxAltitude(): LiveData<Int> {
        return maxAltitude
    }

    fun setMinAltitude(minAltitude: Int) {
        this.minAltitude.value = minAltitude
    }

    fun setMaxAltitude(maxAltitude: Int) {
        this.maxAltitude.value = maxAltitude
    }

    fun setTarget(target: LatLng) {
        this.target.value = target
    }

    fun getTarget(): LiveData<LatLng> {
        return target
    }
}